test/support/protocol_test_case.rb

```diff
@@ -37,11 +41,13 @@ class Detach < StandardError
     # API
 
     def run_protocol_scenario program, dap: true, cdp: true, &scenario
-      write_temp_file(strip_line_num(program))
-      execute_dap_scenario scenario if dap
-      execute_cdp_scenario scenario if cdp
+      Timeout.timeout(30) do
```

Comment:
Why don't we bump the global timeout to 30s? I'm not a fan of the fact that this is hard coded and not configurable.

Reply:
The current `TIMEOUT_SEC` is for individual tests, while the timeout here could have for multiple tests. Forcing them to use the same value will cause problems imo. 
For example, when debugging a hanging test I usually shorten the timeout to 3 to get quicker feedback. But then if I run the entire suite, many of the protocol tests will now fail because of it. So I'd need to change that back to 30 again.